bindings/CMakeLists.txt

```diff
@@ -67,3 +75,41 @@ if(IDYNTREE_USES_PYTHON OR
        MESSAGE(FATAL_ERROR "Swig not found, impossible to compile or generate iDynTree bindings.")
    endif()
 endif()
+
+
+option(IDYNTREE_USES_PYTHON_PYBIND11 "Create iDynTree Python bindings using pybind11" FALSE)
+find_package(pybind11 QUIET)
+if (${IDYNTREE_USES_PYTHON_PYBIND11})
+    if (${pybind11_FOUND})
+        add_subdirectory(pybind11)
+    else()
+       MESSAGE(FATAL_ERROR "pybind11 not found, impossible to generate iDynTree pybind11 bindings.")
+    endif()
+endif()
+
+# Install main __init__.py file.
+if (${IDYNTREE_USES_PYTHON_PYBIND11} OR ${IDYNTREE_USES_PYTHON})
+    if (WIN32)
+        set(NEW_LINE "\n\r")
+    else()
+        set(NEW_LINE "\n")
+    endif()
+    # Clear the file first if it exists.
+    file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/__init__.py "")
+    # If pybind is enabled, add the corresponding import.
+    if (${IDYNTREE_USES_PYTHON_PYBIND11})
+        file(APPEND ${CMAKE_CURRENT_BINARY_DIR}/__init__.py "from . import pybind${NEW_LINE}")
+    endif()
+    # If SWIG is enabled, add the corresponding import.
+    if (${IDYNTREE_USES_PYTHON})
+        file(APPEND "${CMAKE_CURRENT_BINARY_DIR}/__init__.py" "from . import swig${NEW_LINE}")
+        # Create also an alias from swig to the old `bindings` import.
+        file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/bindings.py" "from .swig import *${NEW_LINE}")
+        install(FILES "${CMAKE_CURRENT_BINARY_DIR}/bindings.py"
+                DESTINATION ${PYTHON_INSTDIR})
+        # Add the alias in the __init__.py
+        file(APPEND "${CMAKE_CURRENT_BINARY_DIR}/__init__.py" "from . import bindings${NEW_LINE}")
+    endif()
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/__init__.py"
+             DESTINATION ${PYTHON_INSTDIR})
```

Comment:
This is not exactly what I had in mind but I just realized that the import could not be done as I originally thought.

### What I was thinking

1. Remove the `bindings.py` file
2. Obtain the following `__init__.py` file:

```py
from . import pybind
from . import swig

from . import swig as bindings
```

This, however, does not allow downstream projects to use `import idyntree.bindings` but only `from idyntree import bindings`.

### What we have now

Now the init files and import statements are a bit intertwined and not super clean, but they enable using both import types. I can't see any problem that could emerge from it, and from my side it's ok.

Reply:
Yes. It is exactly as you said. If you want to have the alias and `import idyntree.bindings` you need to have the Python `bindings.py` file.